### ISSUE TITLE
sae_manifold_fit collapses to degenerate null solution on simple 1-harmonic data (assignment mass ~1e-145)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#165]
**gamfit 0.1.123**, `sae_manifold_fit`.

On a clean, low-noise 1-harmonic toy dataset the fit collapses: all atoms are inactive, all assignment mass is numerically zero, and reconstruction R² is negative regardless of schedule kind, `n_iter`, `K`, `alpha`, or `top_k`. This makes the schedule sweep below uninformative because every kind returns the *same* degenerate fit.

Repro (`/tmp/sae_api_schedules.py`):

```python
import numpy as np, gamfit
rng = np.random.default_rng(0)
n = 300
theta = rng.uniform(0, 2*np.pi, size=n)
Z = np.column_stack([np.cos(theta), np.sin(theta),
                     0.3*np.cos(2*theta), 0.3*np.sin(2*theta)]) + 0.05*rng.standard_normal((n,4))

m = gamfit.sae_manifold_fit(Z=Z, K=2, d_atom=1, atom_topology="circle",
                            assignment="ibp", n_iter=20, random_state=0)
print(m.summary())
# {'K':2, 'd_atom':1, 'atom_topology':'circle', 'assignment':'ibp',
#  'reml_score': -691.302, 'reconstruction_r2': -0.0088,
#  'avg_active_atoms': 0.0, 'mean_assignment_mass': 5.6e-146, ...}
```

Sweeping `decay ∈ {geometric, exponential, linear, reciprocal_iter}` and helper functions
`gumbel_geometric_schedule / gumbel_linear_schedule / gumbel_reciprocal_iter_schedule`
all return *bit-identical* fitted reconstructions to one another — including `schedule=None`. The fit never engages the assignment prior, so the schedule is exercised but unobservable.

Expected: on a clean circle dataset, `sae_manifold_fit(K=2, d_atom=1, atom_topology="circle")` should learn a single circular atom with non-trivial assignment mass and positive R².

Likely root cause is the IBP-MAP / softmax assignment initialization underflowing immediately (assignment mass ~1e-146 is consistent with `exp(-large)` at the very first step), but I haven't bisected.

**Comment by SauersML** [comment]
Root cause confirmed and fixed on `main`.

**Two principled root causes, fixed in two commits:**

**1. Periodic latent-dim conflated with harmonic count** — `crates/gam-pyffi/src/lib.rs::sae_build_atom_plans` set `latent_dim = atom_dim` for `Periodic` atoms. With `d_atom >= 2`, that pushes `latent_dim = 2`, but the analytic `PeriodicHarmonicEvaluator` requires `latent_dim == 1`. So `build_sae_basis_evaluators` falls back to the **frozen** `StaticBasisEvaluator` snapshot, and `Phi` / `dPhi/dt` never refresh as coords move during Newton — the kernel has no signal and the assignment posterior drifts into the saturated tail.

Fix in 6b247894: for `Periodic` atoms, pin `latent_dim = 1` and route the user-supplied `d_atom` into `n_harmonics` (basis size = `2·n_harmonics + 1`). Same fix mirrored in the OOS `sae_manifold_predict_oos` plan builder. `sae_manifold_fit_inner` now receives the plan-derived `effective_atom_dim` instead of the raw user value so the optimizer's coord-block width and basis-jacobian latent axis agree with the analytic-evaluator contract.

**2. Python default `learning_rate=0.05` starves IBP-MAP / softmax gradient flow** — the Rust reference unit test `sae_manifold_fit_10_steps_one_harmonic_reaches_high_r2` reaches R² ≥ 0.95 in 10 Newton steps with `learning_rate=1.0` (the natural full damped Gauss-Newton step under Armijo backtracking). At `lr=0.05` the assignment update is so small that the IBP sigmoid drifts into saturation — the `~1e-146` mass you observed.

Fix in 81d77b43: `learning_rate` defaults to `None`; auto-derive at call time — `lr=1.0` for `softmax` / `ibp_map`, `lr=0.05` for `jumprelu` (the hard-gate STE is more sensitive to overshoot). No new CLI flag — the choice follows the assignment prior the caller already supplied. Explicit user overrides still pass through unchanged.

The shipped wheel (`gamfit 0.1.123`) and the bundled `gamfit/_rust.abi3.so` predate both fixes; a fresh `maturin develop` (or the next wheel release) is required for the Python-visible behavior to change.
